pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -100,9 +103,22 @@ func (a *AgentPXEFiles) PersistToFile(directory string) error {
 	if err != nil {
 		return err
 	}
-	err = a.copy(agentVmlinuzFile, fileReader)
-	if err != nil {
-		return err
+
+	if a.cpuArch == arch.RpmArch(types.ArchitectureARM64) {
+		gzipReader, err := gzip.NewReader(fileReader)
+		if err != nil {
+			panic(err)
+		}
+		defer gzipReader.Close()
```

Comment:
This is good, but it makes me notice that we're not actually closing all the fileReaders we open in this function.